### PR TITLE
Fix bad test 01641_memory_tracking_insert_optimize_long

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

See https://clickhouse-test-reports.s3.yandex.net/0/1c472b014d814145023e41cbf437dac5a9dcb494/functional_stateless_tests_(release,_databaseordinary).html#fail1

The test has very tight boundaries that don't look reasonable.

### CODE REVIEW

**Comment by azat** [comment]
>[Functional stateless tests flaky check (address) — Timeout, fail: 2, passed: 43](https://clickhouse-test-reports.s3.yandex.net/23617/301d75dcbe077134873f69c7c3f3356d6d66f69f/functional_stateless_tests_flaky_check_(address).html#fail1)

Right now it fails because of:

```
2021.04.25 08:21:15.866623 [ 291 ] {e046d43d-7fb7-4ae7-ba44-36697502b119} <Information> test_vnhrvt.data_01641 (0dcc5ed8-e1f2-4d06-be9c-be97291e83d2): Cannot OPTIMIZE table: Timeout while waiting for already running merges before running OPTIMIZE with FINAL. Cannot select parts for optimization
```

After #23623 it can be fixed with adjusting `lock_acquire_timeout_for_background_operations`

**Comment by alexey-milovidov** [comment]
@azat The test is too slow.

**Comment by azat** [comment]
>Right now it fails because of:

@alexey-milovidov BTW #23486 will help with investigating such errors

File: tests/queries/0_stateless/01641_memory_tracking_insert_optimize_long.sql [hunk 1]
```diff
@@ -2,20 +2,11 @@ drop table if exists data_01641;
...
--- - is 21MiB if background merges already scheduled
--- - is ~60MiB otherwise
-set max_memory_usage='80Mi';
-system start merges data_01641;
-optimize table data_01641 final;
-
--- definitely should fail
-set max_memory_usage='1Mi';
+-- Definitely should fail and it proves that memory is tracked in OPTIMIZE query.
+set max_memory_usage='10Mi';
```
Comment (azat) [inline tests/queries/0_stateless/01641_memory_tracking_insert_optimize_long.sql:9]:
```suggestion
set max_memory_usage='10Mi', max_untracked_memory=0;
```

**Comment by alexey-milovidov** [comment]
This test is either long or not working. Or it can be made barely working, but I give up.

**Pull request merged**